cluster/src/test/java/io/scalecube/cluster/ClusterTest.java

```diff
@@ -29,9 +39,130 @@ public void testJoinDynamicPort() throws Exception {
     System.out.println("Cluster nodes: " + seedNode.members());
 
     // Shutdown all nodes
-    seedNode.shutdown().get();
-    for (ICluster node : otherNodes) {
-      node.shutdown().get();
+    shutdown(seedNode);
+    shutdown(otherNodes);
+  }
+
+  @Test
+  public void testUpdateMetadata() throws Exception {
+    // Start seed member
+    Cluster seedNode = Cluster.joinAwait();
+
+    // Start member with metadata
+    Map<String, String> metadata = ImmutableMap.of("key1", "value1", "key2", "value2");
+    Cluster metadataNode = Cluster.joinAwait(metadata, seedNode.address());
+
+    // Start other test members
+    int testMembersNum = 10;
+    List<Cluster> otherNodes = new ArrayList<>();
+    for (int i = 0; i < testMembersNum; i++) {
+      otherNodes.add(Cluster.joinAwait(seedNode.address()));
+    }
+
+    // Check all test members know valid metadata
+    for (Cluster node : otherNodes) {
+      Optional<Member> mNodeOpt = node.member(metadataNode.member().id());
+      assertTrue(mNodeOpt.isPresent());
+      Member mNode = mNodeOpt.get();
+      assertEquals(metadata, mNode.metadata());
+    }
+
+    // Subscribe for membership update event all nodes
+    CountDownLatch updateLatch = new CountDownLatch(testMembersNum);
+    for (Cluster node : otherNodes) {
+      node.listenMembership()
+          .filter(MembershipEvent::isUpdated)
+          .subscribe(event -> updateLatch.countDown());
+    }
+
+
+    // Update metadata
+    Map<String, String> updatedMetadata = ImmutableMap.of("key1", "value3");
+    metadataNode.updateMetadata(updatedMetadata);
```

Comment:
If you want to update only single property you need to use method updateMetadataProperty. This method will result in full replacement, so if you use updateMetadata then you need to provide all three properties.